docs/developer/style_guide.md

```diff
@@ -0,0 +1,159 @@
+---
+id: dev_style_guide
+title: Style Guide 
+---
+
+**TL;DR**  - Install the provided pre-commits, follow their recommendations
+
+# General Style 
+Rucio follows [flake8](https://flake8.pycqa.org/en/latest/user/index.html) style, ([with exclusions listed here](https://github.com/rucio/rucio/blob/master/.flake8)). 
+To use them to lint your code, run: 
+```
+python{version} -m pip install flake8 
+flake8 --extend-ignore {codes to ignore} /your/code/path
+```
+
+## Imports
+* Never import using `from x import *`
+* Order alphabetically, then seperated into sections for internal and external dependencies. Group internal imports at the end of the block, and group imports from the same external package.
+* Order modules such that `import {packageA}` is before `from {packageB} import {Module}`
+* Do not import whole packages when single modules would suffice. 
+* Unused imports must be removed. 
+* When a large number of individual imports form a single package/module is required, group them together with `()` and separate them on their own lines. 
+* When importing a module specifically for type checking (e.g. a core module that may not be included in every distribution of rucio, a type from SQLAlchemy), contain them in a block using 
+```python
+from typing import TYPE_CHECKING
+if TYPE_CHECKING: 
+    from {package} import {module}
+```
+
+[`ruff's isort` implementation](https://docs.astral.sh/ruff/faq/#how-does-ruffs-import-sorting-compare-to-isort) handles import sorting in the rucio `pre-commit`s. 
+
+#### Examples: 
+```python 
+ # Wrong
+import rucio  
+from datetime import * 
+import os 
+
+ # Right
+import os 
+from datetime import datetime, timedelta
+
+from rucio.core.did import add_did
+```
+
+```python 
+ # Wrong
+from packageA import moduleA, moduleB, moduleC, moduleD, moduleE, moduleF, moduleG, ...
+
+ # Right
+from packageA import ( 
+  moduleA, 
+  moduleB, 
+  moduleC, 
+  moduleD, 
+  moduleE,
+  moduleF, 
+  moduleG, 
+  ...
+)
+```
+
+# SQLAlchemy Query Guide 
+Rucio uses a custom SQLAlchemy style. 
+Each major clause or operation is given a new line, and a level of indentation is added for each operation within a part of a query.
+
+When queries as executed, perfered output types are either scalars or single values, to avoid using an index to select an element out of a query. 
```

Comment:
```suggestion
When queries are executed, preferred output types are either scalars or single values, to avoid using an index to select an element out of a query. 
```